runtime/plugin/matchparen.vim

```diff
@@ -45,7 +45,7 @@ function! s:Highlight_Matching_Pair()
 
   " Avoid that we remove the popup menu.
   " Return when there are no colors (looks like the cursor jumps).
-  if pumvisible() || (&t_Co < 8 && !has("gui_running"))
+  if pumvisible() || (&term != 'abstract_ui' && &t_Co < 8 && !has("gui_running"))
```

Comment:
Yes and no. While abstract_ui is temporary(it will be removed once the new TUI migration is complete), I'm still not sure how to deal with the `&t_Co` option in vimscript.

Most of the times, this is checked to decide what to display based on the number of terminal colors, but since nvim can have multiple terminals attached, this simply doesn't make sense. 

In the C code where t_Co is checked, I will simply remove the conditions and hardcode the path where most colors are available, but for vimscript I see two choices:

- Add a stub value of 256 to `&t_Co`
- Require that existing vimscript testing for `&t_Co` to be modified to take nvim into consideration.

Both approaches will probably require color schemes designed for multiple UIs to be adapted for nvim to work correctly. For example, in a color scheme that supports terminals with 16, 88 and 256 colors the current solution is something like this:

```vim
if &t_Co == 256
  highlight GROUP ctermfg=x
else if &t_Co == 88
  highlight GROUP ctermfg=y
else
  highlight GROUP ctermfg=z
endif
```

while for nvim the ideal approach would be:

```vim
highlight GROUP term256fg=x term88fg=y term16fg=z
```

Which would display correctly if multiple TUIs with different color count are connected at the same time(each TUI would receive the complete highlight information and use what is appropriate)

Clearly this would require [this change](https://github.com/neovim/neovim/issues/1767#issuecomment-68656372) to work.